interfaces/prompting/prompting.go

```diff
@@ -56,6 +77,93 @@ func (i *IDType) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+type MaxIDMmap []byte
+
+func OpenMaxIDMmap(maxIDFilepath string) (MaxIDMmap, error) {
```

Comment:
these need a doc comment, they also probably belong to an internal package, as it's not something we expect atm some other package to use?  if that was the case we would probably move them somewhere else

Reply:
I'll add the doc comment, thanks.

The problem with it being internal is it needs to be used from both the `interfaces/prompting/requestprompts` and `interfaces/prompting/requestrules` packages, so I'm not sure how to make it accessible to those two but otherwise internal.

Reply:
you put into interfaces/prompting/internal